### PR TITLE
[FLINK-5751] [docs] Add link check script

### PR DESCRIPTION
**Pull request by uce**
This is a slightly adjusted script from @patricklucas as posted in FLINK-5751.

I was wondering whether it makes sense to have it inside the repo, the local work flow would be:
```sh
./build_docs.sh -p
./check_links.sh
```

You can overwrite the URL to check like `./check_links.sh https://ci.apache.org/projects/flink/flink-docs-master/`.

@rmetzger @patricklucas Should we include this?

### CODE REVIEW

File: docs/check_links.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+# Don't abort on any non-zero exit code
+#set +e
```
Comment (patricklucas) [inline docs/check_links.sh]: This is a vestige from me writing this as a Jenkins jobs; you can remove this line and the above comment.

File: docs/check_links.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+target=${1:-"http://localhost:4000"}
+
+# Crawl the docs, ignoring robots.txt, storing nothing locally
+wget --spider -r -nd -nv -e robots=off -p -o spider.log $target
```
Comment (patricklucas) [inline docs/check_links.sh]: $target in quotes

File: docs/check_links.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+# Abort for anything other than 0 and 4 ("Network failure")
+status=$?
+if [ $status -ne 0 ] && [ $status -ne 4 ]; then
+    exit $status
+fi
+
+# Fail the build if any broken links are found
+broken_links_str=$(grep -e 'Found [[:digit:]]\+ broken link' spider.log)
+echo -e "$broken_links_str"
```
Comment (patricklucas) [inline docs/check_links.sh]: If you're going to get rid of my pretty bold-red formatting you can drop the `-e` too. :)

Actually, might as well put the `echo` line inside the `if` too.

**Comment by uce** [comment]
Thanks @patricklucas, I've addressed your comments and re-added the red formatting, which is nice indeed. ;-) I'm going to merge this with you as the author. Hope that's fine.

**Comment by patricklucas** [comment]
:+1:

File: docs/check_links.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+# Abort for anything other than 0 and 4 ("Network failure")
+status=$?
+if [ $status -ne 0 ] && [ $status -ne 4 ]; then
+    exit $status
+fi
+
+# Fail the build if any broken links are found
+broken_links_str=$(grep -e 'Found [[:digit:]]\+ broken link(s)' spider.log)
+if [ -n "$broken_links_str" ]; then
+	echo -e "\e[1;31m$broken_links_str\e[0m"
```
Comment (patricklucas) [inline docs/check_links.sh]: Mixed tabs/spaces in this file if you want to do a no-review follow-up commit.

Comment (uce) [inline docs/check_links.sh]: Thanks for catching this! Fixed in a separate commit.

**Pull request closed without merging**